pkg/reboot/calculator.go

```diff
@@ -23,102 +23,100 @@ const (
 	MaxBatchesAfterFirst      = 10
 )
 
-type SafeTimeCalculator interface {
-	// GetTimeToAssumeNodeRebooted returns the safe time to assume node was already rebooted
-	// note that this time must include the time for a unhealthy node without api-server access to reach the conclusion that it's unhealthy
-	// this should be at least worst-case time to reach a conclusion from the other peers * request context timeout + watchdog interval + maxFailuresThreshold * reconcileInterval + padding
-	GetTimeToAssumeNodeRebooted() time.Duration
-	SetTimeToAssumeNodeRebooted(time.Duration)
-	Start(ctx context.Context) error
-	//IsAgent return true in case running on an agent pod (responsible for reboot) or false in case running on a manager pod
-	IsAgent() bool
+type RebootDurationCalculator interface {
+	// GetRebootDuration returns the safe time to assume node was already rebooted.
+	// Can be either the specified SafeTimeToAssumeNodeRebootedSeconds or the calculated minimum reboot duration.
+	// Note that this time must include the time for a unhealthy node without api-server access to reach the conclusion that it's unhealthy.
+	// This should be at least worst-case time to reach a conclusion from the other peers * request context timeout + watchdog interval + maxFailuresThreshold * reconcileInterval + padding
+	GetRebootDuration(k8sClient client.Client, ctx context.Context, node *v1.Node, operatorNs string) (time.Duration, error)
+	// SetConfig sets the SelfNodeRemediationConfig to be used for calculating the minimum reboot duration
+	SetConfig(config *v1alpha1.SelfNodeRemediationConfig)
 }
 
-type safeTimeCalculator struct {
-	timeToAssumeNodeRebooted, minTimeToAssumeNodeRebooted                   time.Duration
-	wd                                                                      watchdog.Watchdog
-	maxErrorThreshold                                                       int
-	apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout time.Duration
-	log                                                                     logr.Logger
-	k8sClient                                                               client.Client
```

Comment:
What is the advantage of removing the client ? 
IIUC the downside is that it needs to be filtered down in multiple methods now.

Reply:
leftover from an earlier version, where I didn't even initialize the calculator in main, will revert 👍🏼